provider/bikeshed/src/units/essentials.rs

```diff
@@ -18,6 +19,57 @@ use zerovec::ZeroMap;
 
 impl DataProvider<UnitsEssentialsV1Marker> for DatagenProvider {
     fn load(&self, req: DataRequest) -> Result<DataResponse<UnitsEssentialsV1Marker>, DataError> {
+        /// Fills the prefixes map with the SI prefixes (binary and decimal)
+        ///
+        /// # Errors
+        ///
+        /// - `DataError::custom("Invalid prefix: must be Binary or Decimal")` if the prefix is not Binary or Decimal
+        /// - `DataError::custom("Failed to parse pattern key value")` if the pattern key value cannot be parsed
+        fn fill_si(
+            data: &BTreeMap<String, Patterns>,
+            prefixes_map: &mut BTreeMap<PatternKey, String>,
+        ) -> Result<(), DataError> {
+            const BINARY_PREFIX: &str = "1024p";
+            const DECIMAL_PREFIX: &str = "10p";
+
+            data.iter()
+                .filter(|(key, _)| {
+                    key.starts_with(BINARY_PREFIX) || key.starts_with(DECIMAL_PREFIX)
+                })
+                .map(|(key, patterns)| {
+                    let pattern_key_result =
+                        if let Some(trimmed_key) = key.strip_prefix(BINARY_PREFIX) {
+                            trimmed_key.parse::<u8>().map(PatternKey::Binary)
+                        } else if let Some(trimmed_key) = key.strip_prefix(DECIMAL_PREFIX) {
+                            trimmed_key.parse::<i8>().map(PatternKey::Decimal)
+                        } else {
+                            return Err(DataError::custom(
+                                "Invalid prefix: must be Binary or Decimal",
+                            )
+                            .with_debug_context(key));
+                        }
+                        .map_err(|_| {
+                            DataError::custom("Failed to parse pattern key value")
+                                .with_debug_context(&key)
+                        })?;
+
+                    Ok((pattern_key_result, patterns))
+                })
+                .try_for_each(|elem| match elem {
+                    Ok((key, patterns)) => {
+                        if let Some(pattern) = patterns.unit_prefix_pattern.as_ref() {
+                            prefixes_map.insert(key, pattern.to_string());
+                            Ok(())
+                        } else {
+                            Err(DataError::custom("Failed to get pattern").with_debug_context(&key))
+                        }
+                    }
+                    Err(e) => Err(e),
+                })?;
```

Comment:
You should be able to collapse the `map` and `try_for_each` calls

```suggestion
                    if let Some(pattern) = patterns.unit_prefix_pattern.as_ref() {
                        prefixes_map.insert(pattern_key_result, pattern.to_string());
                        Ok(())
                    } else {
                        Err(DataError::custom("Failed to get pattern").with_debug_context(&key))
                    }
                })?;
```